codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/CodegenDelegator.kt

```diff
@@ -175,7 +175,6 @@ open class RustCrate(
  */
 val DefaultPublicModules = setOf(
```

Comment:
I think a better approach is to move `DefaultPublicModules` into the client and server modules, since it doesn't seem like there's a good reason to share it in `codegen-core`. This would also allow us to tailor the module docs to the client/servers separately.